### DISCUSSION TITLE
Is NetTcpBinding working?

### DISCUSSION THREAD

**Discussion by dioptryk** [CoreWCF/CoreWCF#465] [Q&A]
Hello,
I'm researching whether it's possible for us to migrate to CoreWCF, and since we're using NetTcpBinding, this is what I'm testing. It seems there is something wrong with this binding. I tried the samples and got NullReferenceException in:

`echo.WcfInvoke(new NetTcpBinding(), settings.netTcpAddress))`

NetMessageFramingConnectionHandler.cs, line 54. I also tried creating and calling a trivial CoreWCF service from .NET Framework client and got server internal exception on the client (without info). The integration tests in Samples also seem to be commented out... Please advise.

**Comment by birojnayak** [comment]
yes it works both in Linux and Windows, please ensure the port is open and post the stacktrace if there are any more problems... also if it helps, you may look some some test cases here https://github.com/CoreWCF/CoreWCF/blob/main/src/CoreWCF.NetTcp/tests/TransportWithMessageCredentialNetTCPTest.cs

**Comment by dioptryk** [comment]
Thank you for responding. I got it working, my service was registered incorrectly and the DI was unable to resolve it (same issue #458 has, I think). Still, I believe you do have an issue in Samples\DesktopClient\Program.cs. Rather than:

`new Settings().SetDefaults(s_hostname)`

it should say

`new Settings().SetDefaults(s_hostname, "EchoService")`

like it does in NetCoreClient. Then, the tests work correctly.
